src/kernel/mempool_limits.h

```diff
@@ -17,13 +17,33 @@ namespace kernel {
  */
 struct MemPoolLimits {
     //! The maximum allowed number of transactions in a package including the entry and its ancestors.
-    int64_t ancestor_count{DEFAULT_ANCESTOR_LIMIT};
+    uint64_t ancestor_count{DEFAULT_ANCESTOR_LIMIT};
     //! The maximum allowed size in virtual bytes of an entry and its ancestors within a package.
-    int64_t ancestor_size_vbytes{DEFAULT_ANCESTOR_SIZE_LIMIT_KVB * 1'000};
+    uint64_t ancestor_size_vbytes{DEFAULT_ANCESTOR_SIZE_LIMIT_KVB * 1'000};
     //! The maximum allowed number of transactions in a package including the entry and its descendants.
-    int64_t descendant_count{DEFAULT_DESCENDANT_LIMIT};
+    uint64_t descendant_count{DEFAULT_DESCENDANT_LIMIT};
     //! The maximum allowed size in virtual bytes of an entry and its descendants within a package.
-    int64_t descendant_size_vbytes{DEFAULT_DESCENDANT_SIZE_LIMIT_KVB * 1'000};
+    uint64_t descendant_size_vbytes{DEFAULT_DESCENDANT_SIZE_LIMIT_KVB * 1'000};
+
+    /**
+     * @return MemPoolLimits with all the limits set to the maximum
+     */
+    static MemPoolLimits NoLimits()
+    {
+        uint64_t no_limit{std::numeric_limits<uint64_t>::max()};
+        return {no_limit, no_limit, no_limit, no_limit};
+    }
+
+    /** Modify the current mempool limits to reflect what is allowed under CPFP carve out policy. */
+    static MemPoolLimits CPFPCarveOutLimits(const MemPoolLimits& current_limits)
```

Comment:
Okay that makes sense, I didn't have a strong view either way - no problem at all. I've reverted it back to the previous version (calling it `cpfp_carve_out_limits` this time) and then it can be refactored out in a follow-up.